### DISCUSSION TITLE
Light wallet send redirects CAT token to address light wallet can not read

### DISCUSSION THREAD

**Discussion by trgarrett** [Chia-Network/chia-blockchain#9339] [Support]
I've been testing the new CAT support with the beta light wallet, specifically the SBX token. It was my intent to accumulate some CAT in my mostly-offline (near cold) wallet, while I watch and see what the space turns into.

After I received some SBX from the faucet, I sent it on to my cold wallet address: xch...dx2. I was curious to see how the chiaexplorer site supported CAT tokens. When they didn't appear after several hours, I went ahead and added my mostly-offline wallet's keys to the new light wallet. After (quickly) syncing, it could see the tokens as having been received by an xch...5zc address. Since I wanted to purge these sensitive keys from the computer and restore to offline use, I jotted down the address for sending future SBX deposits to, with the expectation that I would track the balance through chiaexplorer.

Something very unexpected happened when I sent additional SBX to the xch...5zc address. The deposit never showed up on the xch...5zc address according to the chiaexplorer. This time, it went through some sort of wormhole and ended up on an xch...ca0 address. After having already purged my "cold" wallet keys from the PC, I decided to re-sync and found these tokens were apparently not recognized by the target wallet as having been received at all.

To troubleshoot, I sent some SBX from my "cold" xch...dx2 wallet back to my primary address of my older wallet. It arrived as expected and showed it came from the xch...5zc address. This was very confusing to me, because it looked like the xch...5zc address must be valid. A send back to the xch...5zc address from my other address resulted in a transaction that would not be added to the balance of the wallet containing the xch...5zc address.

At this point, I'm just trying to understand if I inadvertently burnt my CAT tokens or if I have stumbled on some sort of a bug.

https://www.chiaexplorer.com/blockchain/address/xch1qp4wu7pex90cupkmz0rnca94pavrkvdtnv0s80p2h8qjfsqy9r3qmfsdx2
https://www.chiaexplorer.com/blockchain/address/xch1yjvyt4hwve23mupsq6ccd9n7ztzdlczr79rauchwqzugjnmp3v4qjl35zc
https://www.chiaexplorer.com/blockchain/address/xch1y29nkh4n2ylvg5xk48s84kvezkqrq09q4c77cp8upeqaepm5e0eq5u5ca0

It's not clear to me that whether the xch...ca0 address belongs to my private key or if the "wormhole" it went through completely gave up possession of it, e.g. burned the tokens. If that address does in fact belong to my wallet, I need some sort of path to add it to the valid receive addresses of the wallet such that it will read its balance correctly from the blockchain. I played around with higher values of 'initial_num_public_keys' in the config.yaml in case that might somehow make it aware of the matching address.

I recognize that I probably did something incorrectly and should have just stuck with receive addresses specifically pointed out to me to be used as "receive addresses." I just want to make sure I have the proper understanding of the mistake I made before I potentially handle any more valuable CAT tokens with my wallets.

Thanks for your ideas,
Tim

![image](https://user-images.githubusercontent.com/36970034/142802747-6515e057-df12-4b0f-819a-7b1f6c876570.png)
![image](https://user-images.githubusercontent.com/36970034/142802853-6ba16ffa-b528-406e-83b8-f46263af8002.png)

**Comment by ojura** [comment]
@trgarrett I just submitted the exact same question, although I know it's going on (https://github.com/Chia-Network/chia-blockchain/discussions/9343)

In short: these "incoming" addresses are NOT true wallet receive addresses, but rather the result of Spacebucks CAT wrapped around your true receive address.  By using that address as a true wallet address for sending coins, you wrapped another CAT layer around it.

You are supposed to send coins only to true receive addresses given out by in "Receive Address".

It is unclear if this is recoverable; it would involve constructing a custom puzzle with the proper amount of nested CAT layers. I don't know if they will interfere with each other and if these coins are spendable at all.

**Reply by ftruzzi** [reply]
Thanks @ojura for your explanation. Do you know how to create the CAT-wrapped address starting from the "standard" receive address, given a CAT tail?

**Reply by ojura** [reply]
Hello, sure, you'll need to install the chia python modules and clvm dev tools (run/brun). Define this function in your bash shell:

`cat_puz_hash() {`
`cdv encode $(brun '(a (q 2 30 (c 2 (c 5 (c 23 (c (sha256 28 11) (c (sha256 28 5) ())))))) (c (q (a 4 . 1) (q . 2) (a (i 5 (q 2 22 (c 2 (c 13 (c (sha256 26 (sha256 28 20) (sha256 26 (sha256 26 (sha256 28 18) 9) (sha256 26 11 (sha256 28 ())))) ())))) (q . 11)) 1) 11 26 (sha256 28 8) (sha256 26 (sha256 26 (sha256 28 18) 5) (sha256 26 (a 22 (c 2 (c 7 (c (sha256 28 28) ())))) (sha256 28 ())))) 1))' '(0x72dec062874cd4d3aab892a0906688a1ae412b0109982e1797a170add88bdcdc 0x78ad32a8c9ea70f27d73e9306fc467bab2a6b15b30289791e37ab6e8612212b1 0x'$(cdv decode $1)')' | sed 's/^0x//')`
`}`

(`0x78ad32a8c9ea70f27d73e9306fc467bab2a6b15b30289791e37ab6e8612212b1` is the SBX tail hash; change it as desired, or you can make it a function argument (replace with e.g. `$2`).

Example usage:
`cat_puz_hash xch1svvg2zkdekvjzj0qdwa3uspdlfgq7lztlguf3zrj87zcp3nyhn6sj34t68`
Output: `xch1vnk6f5tm30y7zmyd0v2g2s9xt8fuvm894u82f4l560ttllv7ltsq6ry9u7`

**Reply by ftruzzi** [reply]
Thank you!

**Reply by BekzhanOmirzak** [reply]
Hello, could you help me how to use it ?
